subprojects/kotlin-dsl-tooling-builders/src/crossVersionTest/groovy/org/gradle/kotlin/dsl/tooling/builders/r54/PrecompiledScriptPluginModelCrossVersionSpec.groovy

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright 2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.kotlin.dsl.tooling.builders.r54
+
+import org.gradle.integtests.tooling.fixture.TargetGradleVersion
+import org.gradle.test.fixtures.file.LeaksFileHandles
+import org.gradle.test.fixtures.file.TestFile
+
+import org.gradle.kotlin.dsl.tooling.builders.AbstractKotlinScriptModelCrossVersionTest
+
+import static org.hamcrest.CoreMatchers.hasItem
+import static org.hamcrest.CoreMatchers.startsWith
+import static org.junit.Assert.assertThat
+
+
+@TargetGradleVersion(">=5.4")
+class PrecompiledScriptPluginModelCrossVersionSpec extends AbstractKotlinScriptModelCrossVersionTest {
+
+    @LeaksFileHandles("Kotlin Compiler Daemon working directory")
+    def "given a single project build, the classpath of a precompiled script plugin is the compile classpath of its enclosing source-set"() {
+
+        given:
+        def implementationDependency = withEmptyJar("implementation.jar")
+        def classpathDependency = withEmptyJar("classpath.jar")
+
+        and:
+        withDefaultSettings()
```

Comment:
:x: `withDefaultSettings` was part of the test setup in https://github.com/gradle/gradle/blob/91ed97f81ee638a24915bb20a94db1feb3058eb6/subprojects/kotlin-dsl-test-fixtures/src/main/kotlin/org/gradle/kotlin/dsl/fixtures/AbstractPluginTest.kt#L19

Reply:
On a second though, now that most logic has been moved to `kotlin-dsl-provider-plugins` it shouldn't be a problem. Better to get coverage of all published versions.